fast64_internal/sm64/sm64_objects.py

```diff
@@ -1403,6 +1420,784 @@ def draw_act(self, obj, layout, value):
+
+    @property
+    def obj_name_gfx(self):
+        if self.export_all_selected:
+            return self.gfx_object.name
+        if not self.object_name and not self.gfx_object:
+            return ""
+        else:
+            return self.object_name or self.gfx_object.name
+
+    @property
+    def obj_name_bhv(self):
+        if self.export_all_selected:
+            return self.bhv_object.name
+        if not self.object_name and not self.bhv_object:
+            return ""
+        else:
+            return self.object_name or self.bhv_object.name
+
+    @property
+    def bhv_name(self):
+        return "bhv" + "".join([word.title() for word in toAlnum(self.obj_name_bhv).split("_")])
+
+    @property
+    def geo_name(self):
+        return f"{toAlnum(self.obj_name_gfx)}_geo"
+
+    @property
+    def collision_name(self):
+        return f"{toAlnum(self.obj_name_col)}_collision"
+
+    @property
+    def model_id_define(self):
+        return f"MODEL_{toAlnum(self.obj_name_gfx)}".upper()
+
+    @property
+    def export_level_name(self):
+        if self.level_name == "custom":
+            return self.custom_export_name
+        return self.level_name
+
+    def draw_generic_options(self, layout):
+        split = layout.split(factor=0.333)
+        col = split.column()
+        if self.export_col:
+            box = col.box()
+            box.prop(self, "export_col")
+            box.prop(self, "include_children")
+            box.prop(self, "export_rooms")
+        else:
+            col.prop(self, "export_col")
+        split.prop(self, "export_gfx")
+        if not self.export_all_selected:
+            split.prop(self, "export_bhv")
+        row = layout.row()
+        row.prop(self, "export_all_selected")
+        row.prop(self, "export_script_loads")
+        self.draw_obj_name(layout)
+
+    def draw_level_path(self, layout):
+        if self.export_header_type == "Custom":
+            export_path = f"{toAlnum(self.custom_export_path)}/"
+        else:
+            export_path = f"/levels/{toAlnum(self.level_name)}/"
+        layout.label(text=f"Level export location is: {export_path}")
+
+    def draw_gfx_names(self, layout):
+        if self.export_header_type == "Actor":
+            layout.label(text=f"Objects will export to /actors/{toAlnum(self.obj_name_gfx)}(.c, .h, _geo.c)")
+        else:
+            self.draw_level_path(layout)
+        layout.label(text=f"GeoLayout name will be: {self.geo_name}")
+        layout.label(text=f"Model ID will be: {self.model_id_define}")
+
+    def draw_obj_name(self, layout):
+        split_1 = layout.split(factor=0.45)
+        split_2 = split_1.split(factor=0.45)
+        split_2.label(text="Name")
+        split_2.prop(self, "object_name", text="")
+        if bpy.context.active_object:
+            tmp_obj_name = bpy.context.active_object.name
+        else:
+            tmp_obj_name = ""
+        split_1.label(text=f"or {repr(tmp_obj_name)} if no name")
```

Comment:
if bpy.context.active_object:
```suggestion
        if bpy.context.active_object:
            tmp_obj_name = bpy.context.active_object.name
            split_1.label(text=f"or {repr(tmp_obj_name)} if no name")
```